prow/apis/prowjobs/v1/types.go

```diff
@@ -161,6 +161,15 @@ type ProwJobSpec struct {
 
 	// ReporterConfig holds reporter-specific configuration
 	ReporterConfig *ReporterConfig `json:"reporter_config,omitempty"`
+
+	// Secret specifies if the Job is considered secret
+	// Secret jobs are only shown by deck instances that have the
+	// `--hiddenOnly=true` or `--show-hidden=true` flag set.
+	// This can only be set on Periodics, because we can currently not
+	// filter based on Jobs in Decks tide view
+	// Secrecy on Presubmits/Postsubmits should be configured by setting
+	// the involved repo to hidden instead
+	Secret bool `json:"secret,omitempty"`
```

Comment:
`Secret` is very generic and could be confused with k8s secrets. Please use something like `Hidden` or `HiddenJob` for clarity and consistency with existing terminology.

Reply:
Good point, updated it to `Hidden`